src/Workspaces/CSharp/Portable/CodeGeneration/EnumMemberGenerator.cs

```diff
@@ -25,22 +26,19 @@ internal static EnumDeclarationSyntax AddEnumMemberTo(EnumDeclarationSyntax dest
 
             var member = GenerateEnumMemberDeclaration(enumMember, destination, options, cancellationToken);
 
-            if (members.Count == 0)
-            {
-                members.Add(member);
-            }
-            else if (members.LastOrDefault().Kind() == SyntaxKind.CommaToken)
-            {
-                members.Add(member);
-                members.Add(SyntaxFactory.Token(SyntaxKind.CommaToken));
-            }
-            else
+            if (members.Count > 0 && !members[^1].IsKind(SyntaxKind.CommaToken))
             {
                 var lastMember = members.Last();
                 var trailingTrivia = lastMember.GetTrailingTrivia();
                 members[members.Count - 1] = lastMember.WithTrailingTrivia();
                 members.Add(SyntaxFactory.Token(SyntaxKind.CommaToken).WithTrailingTrivia(trailingTrivia));
-                members.Add(member);
+            }
+
+            members.Add(member);
+
+            if (options.Preferences.Options.GetOption(CSharpCodeStyleOptions.PreferTrailingCommas).Value)
+            {
+                members.Add(SyntaxFactory.Token(SyntaxKind.CommaToken));
             }
```

Comment:
i actually have mixed feelings about this approach.  i actually think the generator should only do this if generating the first item.  if generating a followup item it should take a 'when in rome' aproach where it follows what the existing code does.

Reply:
@CyrusNajmabadi Per my understanding of a previous conversation with @sharwell, code generation should adhere to code-style options when possible. If the user doesn't like the style, he should change it.

Reply:
Some StyleCop Analyzers options have a tri-state true/false/preserve, where preserve means the current state should be followed when adding new items to the same location. The problem with a preserve value is it doesn't allow the user to also express a preference for the behavior when creating a brand new collection.

I lean towards following the defined code style in this case.

Reply:
@sharwell The `false` value for analyzers in almost all code-style options means "preserve" (never report diagnostic). It doesn't enforce the opposite style. It's more of enable/disable rather than a real true/false.